shared/utility/actuation/InverseKinematics.hpp

```diff
@@ -42,177 +42,143 @@ namespace utility::actuation::kinematics {
+        Eigen::Matrix<Scalar, 3, 1> leg_plane_global_z =
+            (is_ankle_above_waist ? -1 : 1) * (Eigen::Matrix<Scalar, 3, 1>::UnitZ() - (cos_z_and_hip_x * hip_x));
+        const Scalar leg_plane_global_z_length = leg_plane_global_z.norm();
+        if (leg_plane_global_z_length > 0) {
+            leg_plane_global_z /= leg_plane_global_z_length;
         }
-        else {
-            NUClear::log<NUClear::DEBUG>(
-                "InverseKinematics::calculateLegJoints : targetLeg and ankleY parallel. This is unhandled at "
-                "the moment. requested pose = \n",
-                target.matrix());
-            return positions;
-        }
-        // Will be unit as ankleY and hipX are normal and unit
-        const Eigen::Matrix<Scalar, 3, 1> legPlaneTangent = ankleY.cross(hipX);
-
-        ankleRoll = std::atan2(ankleX.dot(legPlaneTangent), ankleX.dot(hipX));
-
-        const bool isAnkleAboveWaist = unitTargetLeg.dot(Eigen::Matrix<Scalar, 3, 1>::UnitZ()) < 0;
-
-        const Scalar cosZandHipX   = Eigen::Matrix<Scalar, 3, 1>::UnitZ().dot(hipX);
-        const bool hipRollPositive = cosZandHipX <= 0;
-        Eigen::Matrix<Scalar, 3, 1> legPlaneGlobalZ =
-            (isAnkleAboveWaist ? -1 : 1) * (Eigen::Matrix<Scalar, 3, 1>::UnitZ() - (cosZandHipX * hipX));
-        const Scalar legPlaneGlobalZLength = legPlaneGlobalZ.norm();
-        if (legPlaneGlobalZLength > 0) {
-            legPlaneGlobalZ /= legPlaneGlobalZLength;
-        }
-
-        const Scalar cosHipRoll = legPlaneGlobalZ.dot(Eigen::Matrix<Scalar, 3, 1>::UnitZ());
-        hipRoll = (hipRollPositive ? 1 : -1) * std::acos(std::max(std::min(cosHipRoll, Scalar(1)), Scalar(-1)));
-
-        const Scalar phi4 = M_PI - knee - lowerLeg;
-        // Superposition values:
-        const Scalar sinPIminusPhi2 = std::sin(M_PI - phi2);
-        const Eigen::Matrix<Scalar, 3, 1> unitUpperLeg =
-            unitTargetLeg * (std::sin(phi2 - phi4) / sinPIminusPhi2) + ankleY * (std::sin(phi4) / sinPIminusPhi2);
-        const bool isHipPitchPositive = hipX.dot(unitUpperLeg.cross(legPlaneGlobalZ)) >= 0;
-
-        hipPitch = (isHipPitchPositive ? 1 : -1)
-                   * std::acos(std::max(std::min(legPlaneGlobalZ.dot(unitUpperLeg), Scalar(1)), Scalar(-1)));
-
-        // If leg is above waist then hipX is pointing in the wrong direction in the xy plane
-        Eigen::Matrix<Scalar, 3, 1> hipXProjected = (isAnkleAboveWaist ? -1 : 1) * hipX;
-        hipXProjected.z()                         = 0;
-        hipXProjected.normalize();
-        const bool isHipYawPositive = hipXProjected.dot(Eigen::Matrix<Scalar, 3, 1>::UnitY()) >= 0;
-        hipYaw                      = (isHipYawPositive ? 1 : -1)
-                 * std::acos(std::max(std::min(hipXProjected.dot(Eigen::Matrix<Scalar, 3, 1>::UnitX()), Scalar(1)),
-                                      Scalar(-1)));
+        const Scalar cos_hip_roll = leg_plane_global_z.dot(Eigen::Matrix<Scalar, 3, 1>::UnitZ());
```

Comment:
Dotting a vector with unit z is exactly the same as just accessing the z component
```suggestion
        const Scalar cos_hip_roll = leg_plane_global_z.z();
```
Although maybe this should be angle between too